app/src/test/java/com/invisibleteam/goinvisible/model/TagTypeTest.java

```diff
@@ -0,0 +1,96 @@
+package com.invisibleteam.goinvisible.model;
+
+import com.googlecode.zohhak.api.TestWith;
+import com.googlecode.zohhak.api.runners.ZohhakRunner;
+
+import org.junit.runner.RunWith;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.invisibleteam.goinvisible.model.TagType.REGEXP_DATETIME_STRING;
+import static com.invisibleteam.goinvisible.model.TagType.REGEXP_DATE_STRING;
+import static com.invisibleteam.goinvisible.model.TagType.REGEXP_TEXT_STRING;
+import static com.invisibleteam.goinvisible.model.TagType.REGEXP_TIMESTAMP_STRING;
+import static com.invisibleteam.goinvisible.model.TagType.REGEXP_VALUE_DOUBLE;
+import static com.invisibleteam.goinvisible.model.TagType.REGEXP_VALUE_INTEGER;
+import static junit.framework.Assert.assertEquals;
+
+@RunWith(ZohhakRunner.class)
+public class TagTypeTest {
+
+    @TestWith({
+            "123,       true",
+            "0,         true",
+            "001,       false",
+            "1.0,       false",
+            "1234567,   false"
+    })
+    public void whenValueIntegerRegularExpressionVerifiesText_ReturningCorrectValues(String text, boolean result) {
```

Comment:
whenValueIntegerRegularExpressionVerifiesText_CorrectValuesAreReturned
in all tests, please keep our naming structure